### PR TITLE
Unit tests

### LINKED ISSUE #79: Write unit tests
- [ ] Account
- [ ] Deck
- [ ] Follow
- [ ] Comment
- [ ] Activity
- [ ] All signals

### PR DESCRIPTION
**Pull request by AmirSavand**
Resolves #79

### CODE REVIEW

File: account/tests.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+from django.test import TestCase
+
+from account.models import User, Follow
+
+
+class UserTestCase(TestCase):
+    user: User
```
Comment (Alir3z4) [inline account/tests.py]: Remove this, you're defining it in the setUp already.

File: account/tests.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    def setUp(self):
+        self.user = User.objects.create(
+            username='Amir',
+            email='amir@savandbros.com',
+            password='wow tricky password',
+        )
+
+    def test_user_methods(self):
```
Comment (Alir3z4) [inline account/tests.py]: Bad naming here.
Unit test means testing each unit.
Each method should be tested in individual unit test/method.

File: account/tests.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        self.assertIsNotNone(self.user.joined_since)
+        self.assertFalse(self.user.is_profile_completed)
+
+        self.user.nationality = 'IRAN'
+        self.user.about = 'The meaning of Pure Awesomeness.'
+        self.user.picture = 'http://picture-url-here.com'
+
+        self.assertTrue(self.user.is_profile_completed)
+
+    def test_user_default_values(self):
```
Comment (Alir3z4) [inline account/tests.py]: This is good ;)

File: account/tests.py (hunk 2 above)
Comment (Alir3z4) [inline account/tests.py]: Bad naming here.
Unit test means testing each unit.
Each method should be tested in individual unit test/method.

File: account/tests.py (hunk 1 above)
Comment (Alir3z4) [inline account/tests.py]: Don't need to define here.
You have already declared it in the setUp method.

File: account/tests.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            username='AmirClone',
+            email='amirclone@savandbros.com',
+            password='wow tricky password by amir clone'
+        )
+        
+    def test_user_follow(self):
+        # Make Amir follow Amir Clone
+        follow: Follow = Follow.objects.create(user=self.amir, following=self.clone)
+        
+        self.assertEqual(follow.__str__(), 'Amir follows AmirClone')
```
Comment (Alir3z4) [inline account/tests.py]: Better to be 

```python
self.assertEqual(str(follow), 'Amir follows AmirClone')
```

File: deck/tests.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+from django.test import TestCase
+
+from account.models import User
+from deck.models import Deck, DeckKind, DeckArena
+
+
+class DeckTestCase(TestCase):
+    deck: Deck
```
Comment (Alir3z4) [inline deck/tests.py]: Don't need to define here.
You have already declared it in the setUp method.

**Comment by Alir3z4** [comment]
It's better to remove your actual email addresses as well.
Use dummy email addresses.

File: account/tests.py (hunk 2 above)
Comment (AmirSavand) [reply account/tests.py]: So what to do? Create a user in each unit test function?

File: account/tests.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        self.assertTrue(self.user.is_profile_completed)
+
+    def test_user_default_values(self):
+        self.assertFalse(self.user.is_admin)
+        self.assertFalse(self.user.is_staff)
+        self.assertTrue(self.user.is_active)
+
+
+class FollowTestCase(TestCase):
+    amir: User
```
Comment (Alir3z4) [inline account/tests.py]: You wanna remove this, it's no needed here.
You're defining it at setUp

File: account/tests.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+    clone: User
```
Comment (Alir3z4) [inline account/tests.py]: You wanna remove this, it's no needed here.
You're defining it at setUp

File: deck/tests.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+from account.models import User
+from deck.models import Deck, DeckKind, DeckArena
+
+
+class DeckTestCase(TestCase):
+    def setUp(self):
+        user: User = User.objects.create(
+            username='Amir',
+            email='amir@savandbros.com',
```
Comment (Alir3z4) [inline deck/tests.py]: Remove your email address, use a dummy one.

**Pull request merged**